### PR TITLE
feat(authenticator): Added new slot for confirm sign up

### PR DESCRIPTION
**Pull request by ErikCH**
#### Description of changes
This change adds a new confirm sign up header, form fields.

#### Issue #, if available
#1188 

#### Description of how you validated changes
With updated test.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/e2e/features/ui/components/authenticator/custom-slots.feature [hunk 1]
```diff
@@ -11,14 +11,15 @@ Feature: Custom Slots
...
-  Scenario: Has custom Confirm Sign Up Footer and Header slot text
+  Scenario: Has custom Confirm Sign Up Footer, FormFields and Header slot text
     Given I intercept '{ "headers": { "X-Amz-Target": "AWSCognitoIdentityProviderService.SignUp" } }' with fixture "sign-up-with-email"
     When I click the "Create Account" tab
     When I type a new "email"
     When I type a custom password from label "Password:"
     When I type a custom confirm password from label "Confirm Password:"
     And I click the "Create Account" button
     Then I see "Enter Information:"
+    Then I see "Form Field Information"
```
Comment (ErikCH) [inline packages/e2e/features/ui/components/authenticator/custom-slots.feature:22]: New test to check if the new form field worked

File: docs/src/pages/components/authenticator/confirm-sign-up-fields.angular.mdx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    let-user="user"
+    let-signOut="signOut"
+  >
+    <h2>Welcome, {{ user.username }}!</h2>
+    <button (click)="signOut()">Sign Out</button>
+  </ng-template>
+</amplify-authenticator>
+```
+
+Additionally, you can further customize the page by updating the [headers and footers](http://localhost:5000/components/authenticator?platform=angular#headers--footers).
```
Comment (hbuchel) [inline docs/src/pages/components/authenticator/confirm-sign-up-fields.angular.mdx:18]: Update localhost url?

Comment (ErikCH) [reply docs/src/pages/components/authenticator/confirm-sign-up-fields.angular.mdx:18]: Oops will fix.

**Comment by ErikCH** [comment]
Closing for now.

**Pull request closed without merging**